pandas/_libs/tslibs/offsets.pyx

```diff
@@ -313,6 +313,30 @@ class _BaseOffset(object):
     def __setattr__(self, name, value):
         raise AttributeError("DateOffset objects are immutable.")
 
+    def __eq__(self, other):
+        if is_string_object(other):
+            from pandas.tseries.frequencies import to_offset
```

Comment:
maybe you want to define a function to return this import, we use this in a number of places?

Reply:
Wouldn't this just be trading one one-liner for another?

Reply:
no then then its a function which we can import at the top (and of course if we eventually are able to actually move to_offset to cython its a big win). but its 'cleaner' I think. (certainly can followup on this in new PR). I just don't really like the imports hidden in the functions (as we must do currently), and rather centrailize in a single place.